packages/typescript-estree/src/ts-estree/ts-nodes.ts

```diff
@@ -4,9 +4,9 @@ import type * as ts from 'typescript';
 // Eg: https://github.com/typescript-eslint/typescript-eslint/issues/2388, https://github.com/typescript-eslint/typescript-eslint/issues/2784
 /* eslint-disable @typescript-eslint/ban-ts-comment, @typescript-eslint/prefer-ts-expect-error, @typescript-eslint/no-empty-interface */
 declare module 'typescript' {
-  // @ts-ignore - added in TS 4.5
+  /** @ts-ignore - added in TS 4.5, deprecated and converted to a type-alias in TS 5.3 */
   export interface AssertClause extends ts.Node {}
-  // @ts-ignore - added in TS 4.5
+  /** @ts-ignore - added in TS 4.5, deprecated and converted to a type-alias in TS 5.3 */
   export interface AssertEntry extends ts.Node {}
```

Comment:
https://github.com/microsoft/TypeScript/issues/56477

This is a problem for us.
Testing in the work repo this breaks consumers on TS5.3
Even with the jsdoc style - TS will error on the TS declaration.

![screenshot of type error](https://github.com/typescript-eslint/typescript-eslint/assets/7462525/f41d9f73-4d88-4e75-b46e-ef7d3d1add41)

We don't get this error locally because we use `skipLibChecks`.

Note that this blocks the full support.